File: mycollab-web/src/main/scala/com/mycollab/module/project/view/parameters/ProjectMemberScreenData.scala
package com.mycollab.module.project.view.parameters

import com.mycollab.module.project.domain.ProjectMember
import com.mycollab.module.project.domain.criteria.ProjectMemberSearchCriteria
import com.mycollab.vaadin.mvp.ScreenData

/**
 * @author MyCollab Ltd.
 * @since 5.0.3
 */
object ProjectMemberScreenData {

  class Search(params:ProjectMemberSearchCriteria) extends ScreenData[ProjectMemberSearchCriteria](params) {}

  class Add(params:ProjectMember) extends ScreenData[ProjectMember](params){}

  class InviteProjectMembers extends ScreenData {}

  class Read(params:Any) extends ScreenData[Any](params) {}
}

File: mycollab-mobile/src/main/java/com/mycollab/mobile/module/project/view/parameters/ProjectMemberScreenData.kt
package com.mycollab.mobile.module.project.view.parameters

import com.mycollab.module.project.domain.SimpleProjectMember
import com.mycollab.module.project.domain.criteria.ProjectMemberSearchCriteria
import com.mycollab.vaadin.mvp.ScreenData

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
object ProjectMemberScreenData {
    class Search(param: ProjectMemberSearchCriteria) : ScreenData<ProjectMemberSearchCriteria>(param)

    class InviteProjectMembers() : ScreenData<*>(null)

    class Read(param: Any?) : ScreenData<*>(param)

    class Edit(param: SimpleProjectMember) : ScreenData<SimpleProjectMember>(param)
}